dash/include/dash/halo/iterator/HaloStencilIterator.h

```diff
@@ -0,0 +1,517 @@
+#ifndef DASH__HALO__ITERATOR__HALOSTENCILITERATOR_H
+#define DASH__HALO__ITERATOR__HALOSTENCILITERATOR_H
+
+#include <dash/Types.h>
+
+#include <dash/halo/Halo.h>
+
+#include <vector>
+
+namespace dash {
+
+enum class StencilViewScope : std::uint8_t { INNER, BOUNDARY, ALL };
```

Comment:
Why the restriction to `std::uint8_t`? As this is only a template parameter, it should never occupy memory storage at runtime.

Reply:
Maybe a user wants to use it out of templates.